### ISSUE TITLE
Prisma Crashes Every 15 mins

### ISSUE THREAD

**Issue by mcmar** [prisma/prisma#2926]
**Describe the bug**
I'm running Prisma in ECS using the Fargate template.
The server is crashing every 15 mins with varying log messages that seem to indicate that all networking functionality has been lost.
Different logs messages seem to indicate that network connections are broken and the server becomes unresponsive. Eventually, the load balancer will remove the server and restart it.
The period of the crash seems to vary with traffic. Higher traffic equals more frequent crashes. Lower traffic equals less frequent.

**To Reproduce**
1. Initialize Prisma server from Prisma AWS Fargate template (https://github.com/prismagraphql/prisma-templates/tree/master/aws)
2. Load test the server with a significant amount of traffic, including lots of of websocket subscriptions and server-side subscriptions.
3. Go to ECS cluster (name will be what you used during CF stack creation
4. Click on Prisma service
5. Click on Events tab
6. Observe the number of times that you see the container restart and the downtime between restarts.

**Expected behavior**
The server should be able to operate stably for an extended period of time.

**Screenshots**
Here is the screenshot of the server restarting frequently:
![fireshot capture 1 - amazon ecs_ - https___us-west-2 console aws amazon com_ecs_home](https://user-images.githubusercontent.com/8251430/44167987-8d352e00-a084-11e8-9e95-ecbdc53d2920.png)

**Versions (please complete the following information):**
 - OS: Amazon Linux ECS AMI
 - `prisma` CLI: `prisma/1.13.7 (darwin-x64) node-v8.11.1`
 - Prisma Server: `1.13.7`

**Additional context**
Excerpted log data showing multiple crashes and restarts:
[prisma-crash-logs.txt](https://github.com/prisma/prisma/files/2291406/prisma-crash-logs.txt)

@divyenduz asked me to create the bug here instead of emailing in for support

**Comment by marktani** [comment]
Hey @mcmar, thanks for opening this up. This is unexpected behaviour, so let's find out what causes it so we can identify the underlying issue.

* Do you observe the same behaviour without using server-side subscriptions or websocket subscriptions?
* Are you running a Node server in front of Prisma (for example `graphql-yoga` or `apollo-server`)?
  * If yes, can you reproduce the issue without running those, so by only running queries/mutations or queries/mutations/subscriptions against the Prisma API directly?
* What container size (t2.small, c5.large, ...) do you use for Prisma?
    * How do the metrics look like, for example the memory, CPU and connection graphs?
* What container size (t2.small, c5.large, ...) do you use for your database?
  * How do the metrics look like, for example the memory, CPU and connection graphs?

Basically I am interested in singling out which part of your stack is causing the problem, whether that's the Node server, the Prisma server or the DB server.

**Comment by mcmar** [comment]
It appears that increasing the memory available to our prisma container solved this problem.

The reason that's strange is that Prisma did not log an error saying "Out of memory" or equivalent. Typically the JVM will throw an error if that's the reason for a crash.

Here was our previous memory stats:
- 2GB available for container
- `JVM_OPTS=-Xmx1350m` (1350 MB max-size for JVM heap)
- Container memory utilization between 35%-45% at time of crash (that's pretty low)
- No memory leaks. The memory would increase shortly after bootup, but that's normal. Then it would slowly climb to 35%-45% and level out there. Then after a few hours of stable memory, it would suddenly crash.

Given all of that, it really *doesn't* look like a memory issue, but for some unknown reason, increasing the memory available to the container solved it.

**Comment by dpetrick** [comment]
It doesn't have to be the JVM running out of memory (which would produce the OOM message in the logs). I can recall that we had similar cases where either ECS or Docker decided to kill the container because it went over the configured limit. I would need to look up the details again, but correctly configuring ECS was the fix for us iirc.
